internal/provider/resource_gitlab_user_gpgkey_test.go

```diff
@@ -0,0 +1,285 @@
+var updatedEd25519GPGPubKey string = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mDMEYtzQ6RYJKwYBBAHaRw8BAQdASvO1H8QsiJSN+qmEmBwtgeNi61lXzCmCfUF3
+/5e1qgC0IFRlcnJhZm9ybSA8dGVycmFmb3JtQGdpdGxhYi5jb20+iJAEExYIADgW
+IQTqH8ino1RNUdUDG7H+AGMyH5oPPwUCYtzQ6QIbAwULCQgHAgYVCgkICwIEFgID
+AQIeAQIXgAAKCRD+AGMyH5oPP+pbAP0YgRmwvpipmaBuK4/gDWnE3gO2lR4x+vzL
+ciPmYxshZQD+KAK69/eUwsdbW+uapbtSUNDF4l6PWyKES6qYSUlpswi4MwRi3NDp
+FgkrBgEEAdpHDwEBB0A2d2HUVSFIudz5xegw3HjmH5tDoodbyoURpD7NZhslpojv
+BBgWCAAgFiEE6h/Ip6NUTVHVAxux/gBjMh+aDz8FAmLc0OkCGwIAgQkQ/gBjMh+a
+Dz92IAQZFggAHRYhBEA7LZIX/tW2nvyiCfF9dLzyMph4BQJi3NDpAAoJEPF9dLzy
+Mph4TYEBAPSRfGNlQESyYfUmqV795iFkIgCM5nVzqHHfw0mcH50EAP4qZK9Iobvs
+/yG4eD5jp4nGkRfDlXA+ZsIjmChtIRT2DVy/AQDRFeywZIsvuje6DF1AP7qeqbs+
+dZcWp5qRyFu5zodW2gEAgw7OxfYmlQbz7wwKr3IYnY9MyVp+JwSxyAxZ4X+odAM=
+=XSo3
+-----END PGP PUBLIC KEY BLOCK-----`
+var testEd25519GPGPubKeyForCurrentUser string = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mDMEYtzl7hYJKwYBBAHaRw8BAQdAF8RE2x2wE3w/QJAbB8uVv+9HqYQZJGyNZeLt
+eKyyh2i0IFRlcnJhZm9ybSA8dGVycmFmb3JtQGdpdGxhYi5jb20+iJAEExYIADgW
+IQShzcoy44uIRzOHNnGXiWgn/RdiBQUCYtzl7gIbAwULCQgHAgYVCgkICwIEFgID
+AQIeAQIXgAAKCRCXiWgn/RdiBUDpAQDpcNH5rxVlJS4ATpkFPxbyMuZBdr6eaw2G
+3o8ZPykfbAEAt62eRBhyep+rjPyAHW8YyX4+e+SdCZw/KH0NDbZCGAW4MwRi3OXu
+FgkrBgEEAdpHDwEBB0DE69f1NA/05px8W0ZUkIl0kheMosQ3+UFbt/Jba0DQiojv
+BBgWCAAgFiEEoc3KMuOLiEczhzZxl4loJ/0XYgUFAmLc5e4CGwIAgQkQl4loJ/0X
+YgV2IAQZFggAHRYhBCB2JV+gsGWeld0PcwehRlcCKxpcBQJi3OXuAAoJEAehRlcC
+KxpcSmcBALG/ZILOrVHratA+cZchDQEtOM2rymXdw6AgnllhMj48AQCjWmESbs+C
+GcNNwjo6hAu59BiCvU5+W2of9fpSxBM5CJQ4AQDjyZwDcf6kMu5+bYY9aNcz9skX
+BEMqhn2i2EtNNfH6eAEA1m21vnE8pseBCYHtl9/XJGkB5JH0gUqqRrCf5CAqsgk=
+=/bV6
+-----END PGP PUBLIC KEY BLOCK-----`
+var testEd25519GPGPubKeyWithTrailingNewline string = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mDMEYtzSYhYJKwYBBAHaRw8BAQdAaSePzlrqUWT/hBRqO/oIdfBPh5m57cPwpmjL
+dgUJGcW0IFRlcnJhZm9ybSA8dGVycmFmb3JtQGdpdGxhYi5jb20+iJAEExYIADgW
+IQTFohelsWBqQdLTMuS4BhhIqbpAJQUCYtzSYgIbAwULCQgHAgYVCgkICwIEFgID
+AQIeAQIXgAAKCRC4BhhIqbpAJX8hAP9XCLVIMzO8pleoU82XV+yEVxgKA44uSvat
+zcsIlQrUpwEAkFllbHnquiMxOr+UTXFXIj2L48V2oysbwbhadfIeGwO4MwRi3NJi
+FgkrBgEEAdpHDwEBB0AzJRFF4Ufc8MFl42TDhkMNvMYxVdHogLyOXwK5zoQIo4jv
+BBgWCAAgFiEExaIXpbFgakHS0zLkuAYYSKm6QCUFAmLc0mICGwIAgQkQuAYYSKm6
+QCV2IAQZFggAHRYhBKkMHhbuzwzdKeMiMw7wawPgy8RgBQJi3NJiAAoJEA7wawPg
+y8RgobsA/1uo4Z1ybThL//tNXFeV8J4vr5Kj+1mUO92v4E8oMhL0AQDsWH2H6kn2
+yXEq4d7IEMN2bZOgnIK8q9wcQm7ouYXkDZ2rAP96iz2c2HYWouFZXf0vqzrgJWVB
+ubvgn0HwJkdP6VIUKwEArIF90cFLtKmephxMlwp7h+djEWpVIQ/T31pLYSAH4QA=
+=5+sB
+-----END PGP PUBLIC KEY BLOCK-----
+`
+
+func TestAccGitlabUserGPGKey_basic(t *testing.T) {
+	var key gitlab.GPGKey
+	testUser := testAccCreateUsers(t, 1)[0]
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabUserGPGKeyDestroy,
+		Steps: []resource.TestStep{
+			// Create a user + gpgkey
+			{
+				Config: testAccGitlabUserGPGKeyConfig(testUser.ID, testEd25519GPGPubKey),
```

Comment:
Let's include the config in the test here, see https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008#:~:text=Prefer%20writing%20config%20in%2Dline%20to%20wrapping%20it%20in%20a%20function.

Reply:
Done in 4794d477e21f973f5e85fcb43db159f85d4d164c, thanks!